src/arrayMethodSort.js

```diff
@@ -5,7 +5,33 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    if (compareFunction && typeof compareFunction === 'function') {
```

Comment:
Consider defining a default value for the compare function instead of repeating the loop:

```js
if (!(compareFunction && typeof compareFunction === 'function')) {
  compareFunction = //default compare function; in this case a string comparison
}
for (...) // implement the sort only once
```

This will avoid implementing the same logic twice changing only the condition, and result in a clearer and easier to mantain code